src/Hl7.Fhir.Shims.Base/Specification/Terminology/CodeSystemFilterProcessor.cs

```diff
@@ -0,0 +1,167 @@
+                  ?? throw new CodeSystemUnknownException($"Cannot find codesystem '{codeSystemUri}', so the defined filter(s) cannot be applied.");
+
+            var result = applyFilters(filters, codeSystem);
+
+            return result.Select(c => c.ToContainsComponent(codeSystem, settings));
+        }
+
+        private static List<CodeSystem.ConceptDefinitionComponent> applyFilters(List<ValueSet.FilterComponent> filters, CodeSystem codeSystem)
+        {
+            var result = new List<CodeSystem.ConceptDefinitionComponent>();
+            var properties = codeSystem.Property;
+            bool first = true;
+
+            foreach (var filter in filters)
+            {
+                //If multiple filters are specified within the include, they SHALL all be true. So the second filter, can just filter the result of the first etc.
+                var newResult = first
+                                    ? applyFilter(codeSystem.Concept, properties, filter)
+                                    : applyFilter(result, properties, filter);
+
+                if (first) first = false;
+                result = newResult.ToList();
+            }
+
+            return result;
+        }
+
+        private static IEnumerable<CodeSystem.ConceptDefinitionComponent> applyFilter(List<CodeSystem.ConceptDefinitionComponent> concepts, List<CodeSystem.PropertyComponent> properties, ValueSet.FilterComponent filter)
+        {
+            return filter.Op switch
+            {
+                FilterOperator.IsA => applyIsAFilter(concepts, properties, filter),
+                _ => throw new InvalidOperationException("no filter was selected")
+            };
+        }
+
+        private static IEnumerable<CodeSystem.ConceptDefinitionComponent> applyIsAFilter(List<CodeSystem.ConceptDefinitionComponent> concepts, List<CodeSystem.PropertyComponent> properties, ValueSet.FilterComponent filter)
+        {
+            var result = new List<CodeSystem.ConceptDefinitionComponent>();
+
+            //find descendants based on subsumedBy
+            if (properties.Any(p => p.Code == SUBSUMEDBYCODE))
+            {
+                //first find the parent itself (if it's in the CodeSystem)
+                if (concepts.FindCode(filter.Value) is { } concept)
+                    result.Add(concept);
+
+                //Create a dictionary which lists children by parent.
+                var dict = new Dictionary<string, List<CodeSystem.ConceptDefinitionComponent>>();
+                addConceptsToSubsumedByDict(concepts, dict);
+
+                //find descendants based on that dictionary
+                var descendants = applySubsumedBy(dict, filter);
+                result.AddRange(descendants);
+            }
+            else
+            {
+                //SubsumedBy is not used, we should only check for a nested hierarchy, and include the code and it's descendants
+                if (concepts.FindCode(filter.Value) is { } concept)
```

Comment:
Where are the descendants included? FindCode only includes the code itself?

Reply:
And it's descendants as child codes.